docs/site/Inside-Loopback-Application.md

```diff
@@ -6,81 +6,146 @@ sidebar: lb4_sidebar
+  > Loopback provides extensive features to connect to backend and generate object models from datasource schemas.
 
 The example application [loopback4-example-shopping](https://github.com/strongloop/loopback4-example-shopping)
-is a nice place for developers to start looking on how to use Loopback 4.
+is a nice place for developers to start looking on how to use LoopBack 4.
+
+In the example, `Redis` is used for caching items in the shopping cart and `MongoDB` is used for saving user and order data.
+
+![Shopping Example](https://github.com/strongloop/loopback4-example-shopping/blob/master/example-shopping.png)
+
+In the following sections we will see how LoopBack can be used for the online shopping example.
+
+# Usecases for Online Shopping
+
+For our `Online Shopping` site how the customer might interact with a web or mobile application is the same.
+
+- `Login usecases`
+  - The customer would register as a user.
+  - The customer logs in as a user.
+  - After the user logs in, the mobile/web app creates a shopping cart for the user.
+
+- `Shopping usecases`
+  - User starts adding and removing items to the shopping cart.
+  - If the user logs out, the shopping cart is saved and used for the next user session.
+
+- `Order usecases`
+  - User creates an order with items in the shopping cart.
+  - The mobile/web app submits the order and resets the shopping cart.
+  - User takes a look at all the previous orders.
+
+We have the following API requirements for our `Online Shopping` site from the above usecases:
+
+| Use case              | operations         |  summary             | 
+| --------------------  | --------------------- |----------------------|
+| create a new user     | POST /users           | Basic authentication |
+| query for a user      | GET  /users/{userId}  | query by id          |
+| login as a user       | POST /users/login     | returns a jwt token  |
+| user creates a shopping cart  | PUT /shoppingCarts/{userId}          | create or reset shopping cart  |
+| user adds items to cart       | POST /shoppingCarts/{userId}/items | updates shopping cart  |
+| user removes an item from the cart | DEL /shoppingCarts/{userId}/items/{itemId} | removes an item from cart |
+| user submits the order | POST /users/{userId}/orders | returns order object  |
+| user's list of all orders | GET /users/{userId}/orders | returns all orders linked with user |
+
+- `Recommendation service`
+  - In the example, a [recommender application](https://github.com/strongloop/loopback4-example-shopping/tree/master/packages/recommender) is also defined alongside the `Shopping application` to mock shopping recommendation APIs. This also gives an apt usecase to demonstrate invoking external APIs.
+
+## Exposing the API endpoints
+
+We can now start creating endpoints using the api specs in the above section. Developers typically create server-side "handlers" for each route. The handlers will have to accept incoming request, parse and validate.
```

Comment:
Do we want to use the term `handlers` here? Do you mean `controller functions`?  Handlers aren't a term I've encountered yet in LoopBack 4 with respect to endpoints. If I search `handler` in the LoopBack4 docs, I came across https://loopback.io/doc/en/lb4/apidocs.rest.restbindings.handler.html

![image](https://user-images.githubusercontent.com/6864736/66838486-beccb200-ef32-11e9-9b82-119b88a0de11.png)

But user doesn't have to deal with this when using a controller.

Reply:
`route handlers` is just a common term across node.js community, especially express, to point to server side functions. I will word this better. The idea is to explain the usecase in simple words and then bring in loopback specific terminology.